src/lib/core/scroll/scroll.ts

```diff
@@ -0,0 +1,59 @@
+import {Injectable} from '@angular/core';
+import {Scrollable} from './scrollable';
+import {Subject} from 'rxjs/Subject';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+
+
+/**
+ * Service contained all registered Scrollable references and emits an event when any one of the
+ * Scrollable references emit a scrolled event.
+ */
+@Injectable()
+export class Scroll {
+  /** Subject for notifying that a registered scrollable reference element has been scrolled. */
+  _scrolled: Subject<Event> = new Subject();
+
+  /**
+   * Map of all the scrollable references that are registered with the service and their
+   * scroll event subscriptions.
+   */
+  scrollableReferences: Map<Scrollable, Subscription> = new Map();
+
+  constructor() {
+    // By default, notify a scroll event when the document is scrolled or the window is resized.
+    window.document.addEventListener('scroll', this._notify.bind(this));
```

Comment:
Sounds good. Also something else that I was thinking about: I'm pretty sure that this event will trigger Angular's change detection. It may be worth trying to run it outside Angular's zone, however I'm not sure whether it won't end up updating the view at the proper times.